### DISCUSSION TITLE
[mini.files] Bookmark interaction with mini.clue

### DISCUSSION THREAD

**Discussion by pkazmier** [nvim-mini/mini.nvim#1195] [Q&A] [answered]
Love the new bookmark feature. Very nice. I need some help though. Focusing on a bookmark via `'<char>` does not work for me because I have a mini.clue [trigger for marks](https://github.com/pkazmier/nvim/blob/145e0ed4ef2cadee9fa5c2cabb0939ea0af024d6/lua/plugins/mini/clue.lua#L28). I have to comment out that line in order for this to work. You can see the placeholders for mini.clue triggers in the mini.files help screen:

<img width="1186" alt="image" src="https://github.com/user-attachments/assets/94aa706b-27f6-4810-9993-41722281be13">

**Comment by echasnovski** [comment]
I am afraid that's to be expected and is [the reason is documented in 'mini.clue'](https://github.com/echasnovski/mini.nvim/blob/eb0f557a5797cfe9c720ed84a71582c22bccb4f8/doc/mini-clue.txt#L52): it implements its triggers as buffer-local mappings. Because `g?` is designed to also show which buffer-local mappings user has created, it works as expected.

As the bookmark's `'` is also a buffer-local mapping, it looks like it gets overridden by trigger mapping from 'mini.clue'. However, this is not the default behavior because 'mini.clue' creates its triggers only in listed buffers (i.e. the ones assumed that user wants to keep using for a while). While 'mini.files' buffers (and many other "helper" or "temporary" buffers) are not.

So I assume there is a call to `MiniClue.enable_buf_triggers` either in `FileType` autocommand or 'minifiles.lua' ftplugin. In this case my suggestion is to make sure that trigger for "'" is not created. Maybe something like this (inside 'after/ftplugin/minifiles.lua'):

```lua
local triggers_local, triggers_orig = {}, MiniClue.config.triggers
for _, tr in ipairs(MiniClue.config.triggers) do
  if not (tr.keys == "'" and tr.mode == 'n') then table.insert(triggers_local, tr) end
end

MiniClue.config.triggers = triggers_local
pcall(MiniClue.enable_buf_triggers, vim.api.nvim_get_current_buf())
MiniClue.config.triggers = triggers_orig
```

------

Another approach is to save and restore already present buffer-local mapping for "'", like this:

```lua
local mark_goto = vim.fn.maparg("'", 'n', false, true)
pcall(MiniClue.enable_buf_triggers, vim.api.nvim_get_current_buf())
vim.fn.mapset(mark_goto)
```

However, there is no guarantee that buffer local mappings are already created at the point when filetype plugin is executed (and I am hesitant to adding it). So this might be not the most robust approach.

------

All in all, I'd like to think a bit more of maybe making it a smoother interaction.

**Reply by pkazmier** [reply]
Ah, indeed, you are correct. Here is an auto command I had for mini.files:

```lua
vim.api.nvim_create_autocmd("User", {
  pattern = "MiniFilesBufferCreate",
  callback = function(args)
    local buf_id = args.data.buf_id
    vim.keymap.set("n", "g.", H.toggle_dotfiles, { buffer = buf_id, desc = "Toggle Hidden Files" })
    MiniClue.ensure_buf_triggers(args.data.buf_id)
  end,
})
```

After commenting out that line, everything works fine. I don't know why I added that at some point. Perhaps I wanted mini.clue to work in the mini.files buffers? In any case, I've commented it out now. Thanks!

**Reply by echasnovski** [reply]
Under some angle, it does make certain sense to enable 'mini.clue' in 'mini.files' buffers. As those buffers are meant to be close to "normal" ones. But indeed a special care should be done since introduction of the built-in `'` mapping. I'll think of maybe a better way to handle this.

**Comment by echasnovski** [comment, marked as answer]
After thinking about it over night, it does seem that the current status quo is a good compromise: 'mini.clue' is not automatically enabled in 'mini.files' buffers and if user wants it, adjusting triggers might be needed in order for `'` to work (if it was defined as a trigger globally).

So if anybody encounters similar issue, either [adjust 'mini.clue' triggers before enabling them](https://github.com/echasnovski/mini.nvim/discussions/1195#discussioncomment-10542838) or [not enable at all](https://github.com/echasnovski/mini.nvim/discussions/1195#discussioncomment-10544225).
